### PR TITLE
Add authorisation rules

### PR DESCRIPTION
**Pull request by Baroka-wp**
In this PR we add admin role, and implement the authorization rule. 

- [ ] Install CanCanCan in your project.
- [ ] Add a role column to the users table. 
- [ ] A user can delete a post if it is theirs or if they have an admin role. Use CanCanCan for this authorization.
- [ ] A user can delete a comment if it is theirs or if they have an admin role . Use CanCanCan for this authorization.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -24,6 +24,13 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+    redirect_to user_posts_path(current_user)
+    flash[:success] = 'Post deleted successfully'
+  end
+
```
Comment (AnselemOdims) [inline app/controllers/posts_controller.rb:34]: - [ ] Please kindly ensure you use CanCanCan for authorization in the PostsController.
**Hint:**
  You can include the below line at the top of the PostsControler class
`load_and_authorize_resource`

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Baroka-wp, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Baroka-wp) [reply app/controllers/posts_controller.rb:34]: ## thank you @AnselemOdims

**Review by AnselemOdims — APPROVED** [review]
### Hi  @team 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Baroka-wp** [comment]
# thank you @AnselemOdims for you review

**Pull request merged**